File: frontend/src/api/projects.ts
import { apiClient, type ApiResponse } from './client'

export interface ProjectPayload {
  project_name: string
  industry: string
  sub_industry: string | null
  product: string
  personal_intro: string
  target_audience: string
  platforms: string[]
  current_stage: string
}

export interface Project extends ProjectPayload {
  id: number
  user_id: number | null
  created_at: string
  updated_at: string
}

export async function listProjects(): Promise<Project[]> {
  const response = await apiClient.get<ApiResponse<Project[]>>('/api/projects')
  return response.data.data
}

export async function getProject(projectId: number): Promise<Project> {
  const response = await apiClient.get<ApiResponse<Project>>(`/api/projects/${projectId}`)
  return response.data.data
}

export async function createProject(payload: ProjectPayload): Promise<Project> {
  const response = await apiClient.post<ApiResponse<Project>>('/api/projects', payload)
  return response.data.data
}

export async function updateProject(
  projectId: number,
  payload: Partial<ProjectPayload>,
): Promise<Project> {
  const response = await apiClient.put<ApiResponse<Project>>(`/api/projects/${projectId}`, payload)
  return response.data.data
}

export async function deleteProject(projectId: number): Promise<void> {
  await apiClient.delete<ApiResponse<null>>(`/api/projects/${projectId}`)
}

File: backend/app/api/projects.py
from fastapi import APIRouter, Depends, HTTPException, Query, status
from sqlalchemy.orm import Session

from app.api.dependencies import get_current_user
from app.db.session import get_db
from app.models.user import User
from app.schemas.project import ProjectCreate, ProjectRead, ProjectUpdate
from app.services import project_service


router = APIRouter(prefix="/api/projects", tags=["projects"])


def success_response(data: object, message: str = "") -> dict[str, object]:
    return {"success": True, "data": data, "message": message}


def serialize_project(project: object) -> dict[str, object]:
    return ProjectRead.model_validate(project).model_dump(mode="json")


@router.post("", status_code=status.HTTP_201_CREATED)
def create_project(
    payload: ProjectCreate,
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user),
) -> dict[str, object]:
    project = project_service.create_project(db, payload, user_id=current_user.id)
    return success_response(serialize_project(project), "项目创建成功")


@router.get("")
def list_projects(
    skip: int = Query(default=0, ge=0),
    limit: int = Query(default=100, ge=1, le=200),
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user),
) -> dict[str, object]:
    projects = project_service.get_projects_for_user(db, current_user.id, skip=skip, limit=limit)
    return success_response([serialize_project(project) for project in projects])


@router.get("/{project_id}")
def get_project(
    project_id: int,
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user),
) -> dict[str, object]:
    project = project_service.get_project_for_user(db, project_id, current_user.id)
    if project is None:
        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="项目不存在")

    return success_response(serialize_project(project))


@router.put("/{project_id}")
def update_project(
    project_id: int,
    payload: ProjectUpdate,
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user),
) -> dict[str, object]:
    project = project_service.get_project_for_user(db, project_id, current_user.id)
    if project is None:
        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="项目不存在")

    updated_project = project_service.update_project(db, project, payload)
    return success_response(serialize_project(updated_project), "项目更新成功")


@router.delete("/{project_id}")
def delete_project(
    project_id: int,
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user),
) -> dict[str, object]:
    project = project_service.get_project_for_user(db, project_id, current_user.id)
    if project is None:
        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="项目不存在")

    project_service.delete_project(db, project)
    return success_response(None, "项目删除成功")
